crates/holochain/src/test_utils.rs

```diff
@@ -554,13 +555,16 @@ async fn wait_for_integration_diff<Db: ReadAccess<DbKindDht>>(
 
     let timeout = delay * num_attempts as u32;
 
+    let integration_dump = integration_dump(db).await.unwrap();
+
     panic!(
-        "Consistency not achieved after {:?}ms. Expected {} ops, but only {} integrated. Unintegrated ops:\n\n{}\n{}\n",
+        "Consistency not achieved after {:?}ms. Expected {} ops, but only {} integrated. Unintegrated ops:\n\n{}\n{}\n\n{:?}",
         timeout.as_millis(),
         num_published,
         num_integrated,
         header,
         unintegrated.join("\n"),
+        integration_dump,
```

Comment:
This is super useful. The existing code is great for getting some details about what isn't synced but it doesn't give you a hint about what has happened. Whether the data is missing, failed to validate, or hasn't been properly integrated. In this case I'm seeing the data fail to integrate. That's hopefully a good thing because it means if we work through the validation and integration code it should help resolve the problems with this test and probably others.